src/main/java/org/jboss/netty/handler/ssl/SslHandler.java

```diff
@@ -1655,9 +1655,8 @@ public void run() {
 
                 if (futures != null) {
                     final ClosedChannelException cause = new ClosedChannelException();
-                    final int size = futures.size();
-                    for (int i = 0; i < size; i ++) {
-                        futures.get(i).setFailure(cause);
+                    for (ChannelFuture future : futures) {
```

Comment:
I can undo this particular change and force push the branch, I know there isn't probably any gain with the proposed changes but at least the code is more readable, but indeed not sure if is worth the effort, let me know.